doc/libpmem2/pmem2_map.3.md

```diff
@@ -52,32 +52,17 @@ date: pmem2 API version 1.0
 #include <libpmem2.h>
 
 struct pmem2_config;
+struct pmem2_source;
 struct pmem2_map;
-int pmem2_map(const struct pmem2_config *config, struct pmem2_map **map_ptr);
+int pmem2_map(const struct pmem2_config *config, const struct pmem2_source *source,
+    struct pmem2_map **map_ptr);
 ```
 
 # DESCRIPTION #
 
 The **pmem2_map**() function creates a new mapping in the virtual address space
-of the calling process. This function requires that following arguments are set
-in the *config* object:
-
-* The source of the mapping, which may be either an opened file descriptor or
-an opened handle to a file.
-
-    * The file descriptor has to be opened using *O_RDONLY* or *O_RDWR* file
-access mode and passed via **pmem2_config_set_fd**(3). Note on Windows the
-provided file descriptor is converted to a handle using **_get_osfhandle**().
-For details please see the **open**(3), **pmem2_config_set_fd**(3) and
-**_get_osfhandle**() manual pages.
-
-    * Similarly, the handle has to be created with an access mode of
-*GENERIC_READ* or *(GENERIC_READ | GENERIC_WRITE)* and passed via
-**pmem2_config_set_handle**(3). For details please see the **CreateFile**() and
-**pmem2_config_set_handle**(3) manual pages. (Windows only)
```

Comment:
Please move this to the relevant man page instead of removing this.

Reply:
These things are mostly repeated in source_from_fd already. The only info that was missing was about the windows generic read/write thing - I added that.